src/_pytest/assertion/rewrite.py

```diff
@@ -291,12 +293,12 @@ def _read_pyc(source, pyc, trace=lambda x: None):
     Return rewritten code if successful or None if not.
     """
     try:
-        fp = open(pyc, "rb")
+        fp = open(str(pyc), "rb")
```

Comment:
Ideally we should use `os.fspath`, but py36+ only. 😬

Reply:
Can be done for it then already, no?

Reply:
What do you mean? We still need to support py35.

Reply:
Hmmm probably it pays of to create a `fspath`-like function in `compat`... 🤔